File: src/algorithms/stacks-queues/stack-design/min-stack/sources/min-stack_test.py
import importlib

mod = importlib.import_module("min-stack")
min_stack = mod.min_stack

assert min_stack([5, 3, 7, 1, 8]) == 1
assert min_stack([1, 2, 3]) == 1
assert min_stack([3, 2, 1]) == 1
assert min_stack([42]) == 42
assert min_stack([7, 7, 7]) == 7
assert min_stack([5, -3, 2, -1]) == -3
assert min_stack([1, 5, 10, 20]) == 1
assert min_stack([20, 10, 5, 1]) == 1

if __name__ == "__main__":
    print("All tests passed!")

File: src/algorithms/stacks-queues/stack-design/min-stack/__tests__/min-stack_test.rs
include!("../sources/min-stack.rs");

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn default_example() {
        assert_eq!(min_stack(&[5, 3, 7, 1, 8]), 1);
    }

    #[test]
    fn ascending_sequence() {
        assert_eq!(min_stack(&[1, 2, 3]), 1);
    }

    #[test]
    fn descending_sequence() {
        assert_eq!(min_stack(&[3, 2, 1]), 1);
    }

    #[test]
    fn single_element() {
        assert_eq!(min_stack(&[42]), 42);
    }

    #[test]
    fn all_equal() {
        assert_eq!(min_stack(&[7, 7, 7]), 7);
    }

    #[test]
    fn negative_numbers() {
        assert_eq!(min_stack(&[5, -3, 2, -1]), -3);
    }

    #[test]
    fn minimum_first() {
        assert_eq!(min_stack(&[1, 5, 10, 20]), 1);
    }

    #[test]
    fn minimum_last() {
        assert_eq!(min_stack(&[20, 10, 5, 1]), 1);
    }
}
